### PR TITLE
Add missing OnHashChange notification

### PR DESCRIPTION
**Pull request by dcherniev**
Added OnHashChange notification which should be sent when during UnsubscribeVehicleData 2 applications are registered;
Added expectation of new notification to update UnsubscribeVehicleRequestTest unit tests.

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review: @LuxoftAKutsan, @okozlovlux

**Comment by LuxoftAKutsan** [comment]
Are you sure that this PR should be send to develop?

**Comment by dcherniev** [comment]
@LuxoftAKutsan Rebased PR changes onto correct branch.

File: src/components/application_manager/test/commands/mobile/unsubscribe_vehicle_request_test.cc [hunk 1]
```diff
@@ -192,7 +192,7 @@ void UnsubscribeVehicleRequestTest::UnsubscribeSuccessfully() {
   application_set_.insert(mock_app);
   DataAccessor<am::ApplicationSet> accessor(application_set_, app_set_lock_);
   EXPECT_CALL(app_mngr_, application(kConnectionKey))
-      .WillOnce(Return(mock_app));
+      .WillRepeatedly(Return(mock_app));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/unsubscribe_vehicle_request_test.cc]: @dcherniev please use ON_CALL

Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/unsubscribe_vehicle_request_test.cc]: @LuxoftAKutsan Fixed, please check 9efc8bf.

**Comment by dcherniev** [comment]
@LuxoftAKutsan Ok

**Pull request merged**